src/librustc/ty/layout.rs

```diff
@@ -523,8 +523,12 @@ pub type FieldPath = Vec<u32>;
 /// A structure, a product type in ADT terms.
 #[derive(PartialEq, Eq, Hash, Debug)]
 pub struct Struct {
+    /// Maximum alignment of fields and repr alignment.
     pub align: Align,
 
+    /// Alignment of struct fields, for comparison with LLVM type.
+    pub abi_align: Align,
```

Comment:
Naming is hard :) The purpose off this field is to keep track of what LLVM thinks the struct's alignment is. This is used in a few different ways:
* Calculating manual field padding when Rust and LLVM have different ideas about alignment
* Telling alloca alignment of over aligned types (I wanted the be conservative and only specify alloca alignment when they the type is over aligned)
* Sanity checking Rust and LLVM type sizes and alignment in https://github.com/bitshifter/rust/blob/6b9c5d1ff6df1e0d6d181fb5049c3108359b47ee/src/librustc_trans/type_of.rs#L120-L133

Originally I called this field `base_align` which you also suggested in another comment, so perhaps I should go back to that. Other options might be `llvm_align`, `default_align`, `orig_align`.